feature-cloud-backup-impl/src/main/java/io/novafoundation/nova/feature_cloud_backup_impl/data/serializer/CloudBackupSerializer.kt

```diff
@@ -58,13 +58,23 @@ internal class JsonCloudBackupSerializer() : CloudBackupSerializer {
 
     override suspend fun deserializePublicData(backup: ReadyForStorageBackup): Result<SerializedBackup<EncryptedPrivateData>> {
         return runCatching {
-            gson.fromJson(backup.value)
+            gson.fromJson<SerializedBackup<EncryptedPrivateData>>(backup.value).also {
+                // Gson doesn't fail on missing fields so we do some preliminary checks here
+                requireNotNull(it.publicData)
+                requireNotNull(it.privateData)
+
+                // Do not allow empty backups
+                require(it.publicData.wallets.isNotEmpty())
```

Comment:
In this case we will receive IllegalArgumentException, maybe it's better to throw CorruptedBackupException in this case

Reply:
It will be transformed to the CorruptedBackupException on the upper level by the service. Basically Service is the one who transforms errors